.NET/Microsoft.Recognizers.Definitions.Common/German/DateTimeDefinitions.cs

```diff
@@ -48,6 +48,11 @@ public static class DateTimeDefinitions
       public const string WrittenMonthRegex = @"((monat\s*)?(?<month>apr(il|\.)|aug(ust|\.)|dez(ember|\.)|feb(ruar|ber|\.)|j[äa]n(uar|ner|\.)|jul(e?i|l\.)|jun([io]|\.)|märz|mai|nov(ember|\.)|okt(ober|\.)|sept?(ember|\.)))";
       public static readonly string MonthSuffixRegex = $@"(?<msuf>(im\s*|des\s*)?({RelativeMonthRegex}|{WrittenMonthRegex}|{MonthNumRegex}))";
       public const string DateUnitRegex = @"(?<unit>jahre|jahr|monate|monat|wochen?|tage|tag)(s)?";
+      public const string HalfTokenRegex = @"^(halb)";
+      public const string QuarterToTokenRegex = @"^(viertel\s+vor|dreiviertel)\s*$";
+      public const string QuarterPastTokenRegex = @"^(viertel\s+nach)\s*$";
+      public const string ThreeQuarterToTokenRegex = @"^(viertel|dreiviertel\s+vor)\s*$";

```

Comment:
"dreiviertel vor" isn't something you'd use in German, but it's probably a nice idea to add it because non-native speakers might become a little creative with forming these expressions, maybe.